### PR TITLE
fix: remove typesVersions

### PR DESCRIPTION
**Pull request by weboko**
## Problem

`@waku/dns-discovery`'s types are not visible in `web-chat` example. 

## Solution

Remove `typeVersions` since it is not needed. 

## Notes

- Related to https://github.com/waku-org/js-waku/issues/1124
- Links: [TS docs](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#version-selection-with-typesversions)

### CODE REVIEW

File: packages/dns-discovery/package.json [hunk 1]
```diff
@@ -10,13 +10,6 @@
       "import": "./dist/index.js"
     }
   },
-  "typesVersions": {
-    "*": {
-      "*": [
-        "dist/*"
```
Comment (weboko) [inline packages/dns-discovery/package.json:16]: Originally I found out that if changing it to `./dist/*.d.ts` will fix the problem as well - my assumption it is because typescript need to have specific path / pattern to point only to `.d.ts` files. Strangely I see this mistake in docs so not sure. 

But since `typesVersions` are not needed and `types` will be used otherwise I am just removing this section.

Comment (weboko) [reply packages/dns-discovery/package.json:16]: I tried to test it in other scenarios and couldn't find any edge cases.

Comment (weboko) [reply packages/dns-discovery/package.json:16]: I think the was introduced in these PRs:
- https://github.com/waku-org/js-waku/pull/892
- https://github.com/waku-org/js-waku/pull/1097

This shouldn't be a problem for current changes.
